src/antsibull_core/pydantic.py

```diff
@@ -0,0 +1,84 @@
+# Author: Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Ansible Project, 2024
+
+"""
+Helpers for pydantic.
+"""
+
+from __future__ import annotations
+
+import inspect
+import typing as t
+from collections.abc import Collection
+
+import pydantic as p
+
+
+def _is_basemodel(a_type: t.Any) -> bool:
+    try:
+        return inspect.isclass(a_type) and issubclass(a_type, p.BaseModel)
```

Comment:
True. It would be nicer to avoid the exception on Python 3.11+, but as long as there's the awkward behavor with Python 3.9/3.10 the exception should be fine.

Reply:
(Fixed again in e210e0d401333cd0a81b1e39d940d1b30f4c6ad3...)